goalie/mesh_seq.py

```diff
@@ -460,17 +493,81 @@ def _reinitialise_fields(self, initial_conditions):
             )
 
     @PETSc.Log.EventDecorator()
-    def get_checkpoints(self, solver_kwargs=None, run_final_subinterval=False):
+    def _solve_forward(self, update_solutions=True, solver_kwargs=None):
         r"""
-        Solve forward on the sequence of meshes, extracting checkpoints corresponding
-        to the starting fields on each subinterval.
+        Solve a forward problem on a sequence of subintervals. Yields the final solution
+        on each subinterval.
 
+        :kwarg update_solutions: if ``True``, updates the solution data
+        :type update_solutions: :class:`bool`
         :kwarg solver_kwargs: parameters for the forward solver
-        :type solver_kwargs: :class:`dict` with :class:`str` keys and values which may
-            take various types
+        :type solver_kwargs: :class:`dict` whose keys are :class:`str`\s and whose values
+            may take various types
+        :returns: the solution data of the forward solves
+        :rtype: :class:`~.ForwardSolutionData`
+        """
+        # TODO docstring tag for yield?
```

Comment:
Not sure if it's standard practice but you could put
```
:yields: ...
:ytype: ...
```

Reply:
Thanks! Done in c0abbab024ffc394978cef64dfc56b464adb2132